src/Native/System.Native/pal_networking.cpp

```diff
@@ -1929,29 +1929,29 @@ static void ConvertFdSetPlatformToPal(FdSet& palSet, fd_set& platformSet, int32_
 #if !HAVE_FDS_BITS && !HAVE_PRIVATE_FDS_BITS
     for (int i = 0; i < fdCount; i++)
     {
-        int word = i / static_cast<int>(PAL_FDSET_NFD_BITS);
-        int bit = i % static_cast<int>(PAL_FDSET_NFD_BITS);
-        if ((palSet.Bits[word] & (1 << bit)) == 0)
+        uint32_t* word = &palSet.Bits[i / static_cast<int>(PAL_FDSET_NFD_BITS)];
+        uint32_t mask = 1 << (i % static_cast<int>(PAL_FDSET_NFD_BITS));
+
+        if (FD_ISSET(i, &platformSet))
         {
-            FD_SET(i, &platformSet);
+            *word |= mask;
         }
         else
         {
-            FD_CLR(i, &platformSet);
+            *word &= ~mask;
         }
     }
 #else
-
     size_t bytesToCopy = static_cast<size_t>((fdCount / 8) + ((fdCount % 8) == 0 ? 1 : 0));
 
-    uint8_t* dest;
+    uint8_t* source;
```

Comment:
Can you remind me why we have this whole else section?  Is it purely a performance thing to be able to use memcpy, and we're sure that's valid?